packages/headless/src/api/knowledge/stream-answer-api.ts

```diff
@@ -0,0 +1,287 @@
+import {
+  ArrayValue,
+  BooleanValue,
+  RecordValue,
+  Schema,
+  StringValue,
+} from '@coveo/bueno';
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {
+  GeneratedAnswerStyle,
+  GeneratedContentFormat,
+} from '../../features/generated-answer/generated-response-format';
+import {selectPipeline} from '../../features/pipeline/select-pipeline';
+import {selectQuery} from '../../features/query/query-selectors';
+import {QueryState} from '../../features/query/query-state';
+import {selectSearchHub} from '../../features/search-hub/search-hub-selectors';
+import {
+  ConfigurationSection,
+  DebugSection,
+  GeneratedAnswerSection,
+  SearchSection,
+} from '../../state/state-sections';
+import {GeneratedAnswerCitation} from '../generated-answer/generated-answer-event-payload';
+import {SearchRequest} from '../search/search/search-request';
+import {answerSlice} from './answer-slice';
+
+type StateNeededByAnswerAPI = ConfigurationSection &
+  GeneratedAnswerSection &
+  SearchSection &
+  DebugSection & {answer: ReturnType<typeof answerApi.reducer>};
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+interface GeneratedAnswerStream {
+  answerStyle: GeneratedAnswerStyle | undefined;
+  contentFormat: GeneratedContentFormat | undefined;
+  answer: string | undefined;
+  citations: GeneratedAnswerCitation[] | undefined;
+  generated: boolean;
+  isStreaming: boolean;
+  isLoading: boolean;
+}
+interface HeaderMessage {
+  answerStyle: GeneratedAnswerStyle;
+  contentFormat: GeneratedContentFormat;
+}
+
+type PayloadType =
+  | 'genqa.headerMessageType'
+  | 'genqa.messageType'
+  | 'genqa.citationsType'
+  | 'genqa.endOfStreamType';
+
+const headerMessageSchema = new Schema<HeaderMessage>({
+  answerStyle: new StringValue(),
+  contentFormat: new StringValue(),
+});
+
+const messageSchema = new Schema({
+  textDelta: new StringValue(),
+});
+
+const citationsSchema = new Schema({
```

Comment:
Hum, interesting that all the endpoint output gets validated with Bueno... 🤔 

Typically we use it to validate implementers code (ie: validate arguments types/shape that they pass to Headless functions), and not what our API return.

I can see it will make it easier to "detect that something went horribly wrong from the API", but.... we'll still just end up showing an irrecoverable error to the end user, just in a different manner.

And nothing that an implementer/client can do to fix if that happen either, except wait for us to fix.

So, all in all, interesting, but I'm dubious about what tangible improvement it will bring to our users.

Reply:
I share your uncertainty about the dubious value it will add to our end user. 

Being new working in the ui-kit, after seeing this comment https://github.com/coveo/ui-kit/pull/4014#discussion_r1654943794 I just thought I would overkill it. 

I welcome any suggestion to make the type check more relevant.